File: src/main/java/gg/mineral/server/network/packet/play/clientbound/TimeUpdatePacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.server.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public record TimeUpdatePacket(long ageOfWorld, long timeOfDay) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeLong(ageOfWorld);
        os.writeLong(timeOfDay);
    }

    @Override
    public byte getId() {
        return 0x03;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/TimeUpdatePacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class TimeUpdatePacket(val ageOfWorld: Long, val timeOfDay: Long) : Packet.Outgoing {
    override fun serialize(os: ByteBuf) {
        os.writeLong(ageOfWorld)
        os.writeLong(timeOfDay)
    }

    override val id: Byte
        get() = 0x03
}
